internal/ad/ad.go

```diff
@@ -448,16 +447,32 @@ func (ad *AD) parseGPOs(ctx context.Context, gpos []gpo, objectClass ObjectClass
 
 			log.Debugf(ctx, "Parsing GPO %q", name)
 
-			class := "User"
+			// We need to consider the uppercase version of the name as well,
+			// which could occur in some of the default GPOs such as Default
+			// Domain Policy.
+			classes := []string{"User", "USER"}
 			if objectClass == ComputerObject {
-				class = "Machine"
+				classes = []string{"Machine", "MACHINE"}
 			}
-			f, err := os.Open(filepath.Join(ad.sysvolCacheDir, "Policies", filepath.Base(url), class, "Registry.pol"))
-			if errors.Is(err, fs.ErrExist) {
-				return err
-			} else if errors.Is(err, fs.ErrNotExist) {
+
+			// Store both capitalized (common case) and uppercase (edge case)
+			// errors if needed.
+			var err, upcaseErr error
+			var f *os.File
+			for _, class := range classes {
+				f, upcaseErr = os.Open(filepath.Join(ad.sysvolCacheDir, "Policies", filepath.Base(url), class, "Registry.pol"))
```

Comment:
Actually, rereading, I’m thinking of a better way with a local error.

1. You have out of scope "global" error `err`.
2. You have a `var e error` inside the loop, this is the one you affect to os.Open.
3. If err == nil and e != nil -> assign e to err
4. else if e == nil -> assign err to nil.
5. Then, outside of the for, you can check the value of `err`, which will be the first error (or reset to nil if the second loop range worked)

I think this will make the whole thing more readable as you check for an error and print in debug log the error you actually checked.

Does it make sense?

Reply:
That was exactly it! I find it more future-proof for us to avoid fixing the intentional mismatch we had between the if clause and the debug message printed.